common/scripts/generate-api-diff.mjs

```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/env node
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * This script is used to generate api.json files for beta features for uploading to apiview.dev for review.
+ * This generates two files:
+ *   1. baseline.api.json - the api.json file with the beta feature removed
+ *   2. feature.api.json - the api.json file with the beta feature enabled
+ *
+ * @example
+ * `node ./generate-api-diff.mjs --feature nameOfFeature`
+ */
+
+import { exit } from 'process';
+import yargs from 'yargs/yargs';
+import featureDefinitions from '../config/babel/features.js';
+import { exec } from './lib/exec.mjs';
+import fs from 'fs';
+import path from 'path';
+import { fileURLToPath } from 'url';
+import { dirname } from 'path';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(__filename);
+
+const destinationDir = path.join(__dirname, '../../apis');
+
+async function main() {
+  // Clean any existing results first
+  if (fs.existsSync(destinationDir)) {
+    fs.rmSync(destinationDir, { recursive: true });
+  }
+
+  const { feature } = parseArgs(process.argv);
+  const { features, inProgressFeatures, stabilizedFeatures } = featureDefinitions;
+
+  if (stabilizedFeatures.includes(feature)) {
+    console.error(`ERROR: Feature ${feature} is already stabilized. Please use a beta feature.`);
+    exit(-1);
+  }
+
+  if (!features.includes(feature)) {
+    console.error(`ERROR: Could not find beta feature "${feature}" in features.js file.`);
+    exit(-1);
+  }
+
+  // before making changes to the feature definitions file, make a copy of it
+  const featureDefinitionsFile = path.join(__dirname, '../config/babel/features.js');
+  fs.copyFileSync(featureDefinitionsFile, `${featureDefinitionsFile}.bak`);
+
+  const isAlphaFeature = inProgressFeatures.includes(feature);
+
+  if (isAlphaFeature) {
+    await exec('rush switch-flavor:beta-release');
+  } else {
+    await exec('rush switch-flavor:stable');
+  }
+
+
+  modifyFeatureDefinitionsFile(feature, isAlphaFeature ? 'alphaToBeta' : 'betaToStable');
+  console.log(`Generating baseline.api.json file`);
+  const baselineFilePath = await generateApiFile(false);
+
+  modifyFeatureDefinitionsFile(feature, isAlphaFeature ? 'betaToAlpha' : 'stableToBeta');
+  console.log(`Generating feature.api.json file`);
+  const featureFilePath = await generateApiFile(true);
+
+  // restore the feature definitions file
+  fs.copyFileSync(`${featureDefinitionsFile}.bak`, featureDefinitionsFile);
+  fs.rmSync(`${featureDefinitionsFile}.bak`);
+
+  // restore the flavor
+  await exec('rush switch-flavor:beta');
```

Comment:
Maybe not for this PR, it would be nice to check which flavor was selected and use it here instead

Reply:
I like that! Sounds doable, I'll try add that

Reply:
Updated! Turns out we have a `getBuildFlavor()` fn already for using in our scripts! Great suggestion